service/dap/server_test.go

```diff
@@ -2713,6 +2706,14 @@ func TestBadLaunchRequests(t *testing.T) {
 
 		client.LaunchRequestWithArgs(map[string]interface{}{"mode": "debug", "program": fixture.Source, "buildFlags": "bad flags"})
 		expectFailedToLaunchWithMessage(client.ExpectErrorResponse(t), "Failed to launch: Build error: exit status 1")
+		client.LaunchRequestWithArgs(map[string]interface{}{"mode": "debug", "program": fixture.Source, "noDebug": true, "buildFlags": "bad flags"})
```

Comment:
you could also add a test above where noDebug is something other than a bool

Reply:
noDebug other than a bool will be ignored (not a bad request) and the launch requests should work as if it wasn't set. I see the other way - failing on non-bool type value - isn't wrong either. I am not too worried about it right now. If you want, feel free to add extra check and make it to the bad request case.
Added to the launch default test for now.

Reply:
Sure, looks good. I just meant we needed a test "documenting" behavior, whatever it tis.  
I did the same thing for a number of other flags - if they are bad, don't generate a launch error, but quietly use a default. Suzy and I had a discussion about this because she wanted to use the bad launch approach:
https://github.com/go-delve/delve/pull/2379#issuecomment-806053504